amethyst_ui/src/pass.rs

```diff
@@ -297,6 +293,16 @@ impl Pass for DrawUi {
                 }
                 // Build text sections.
                 let editing = editing.get(entity);
+                let rendered_string = if ui_text.password {
+                    // Build a string composed of black dot characters.
+                    let mut ret = String::with_capacity(ui_text.text.len());
+                    for _grapheme in ui_text.text.graphemes(true) {
```

Comment:
Could just do `ui_text.text.graphemes(true).map(|_| 'u{2022}').collect()`

Reply:
This has the disadvantage that the string isn't allocated with the correct capacity, thus potentially causing more allocations as the string gets resized.